modernizer-annotation-processor/src/main/java/org/gaul/modernizer_annotation_processor/ModernizerAnnotationProcessor.java

```diff
@@ -0,0 +1,331 @@
+        return getClassHeader(classElement).replace("$", "\\$") + "(\\$.+)?";
+    }
+
+    private String getClassHeader(Element classElement) {
+        String packageName = processingEnv
+            .getElementUtils()
+            .getPackageOf(classElement)
+            .getQualifiedName()
+            .toString()
+            .replace('.', '/');
+        String packagePrefix = !packageName.isEmpty() ? packageName + "/" : "";
+        String classHeader = packagePrefix + getFullClassName(classElement);
+        return classHeader;
+    }
+
+    private String getFullClassName(Element classElement) {
+        List<String> parentClasses = new ArrayList<String>();
+        Element enclosingElement = classElement.getEnclosingElement();
+        while (enclosingElement != null &&
+            enclosingElement.getKind().isClass()) {
+            parentClasses.add(enclosingElement.getSimpleName().toString());
+            enclosingElement = enclosingElement.getEnclosingElement();
+        }
+        StringBuilder className = new StringBuilder();
+        for (int index = parentClasses.size() - 1; index >= 0; index--) {
+            className.append(parentClasses.get(index));
+            className.append("$");
+        }
+        return className.toString() + classElement.getSimpleName();
+    }
+
+    private String getMethodIdentifierString(Element methodElement) {
+        TypeVisitor<ExecutableType, Void> executableTypeVisitor =
+            new SimpleTypeVisitor6<ExecutableType, Void>() {
+                @Override
+                public ExecutableType visitExecutable(
+                    ExecutableType executableType, Void obj
+                ) {
+                    return executableType;
+                }
+            };
+        ExecutableType method =
+            methodElement.asType().accept(executableTypeVisitor, null);
+        List<? extends TypeMirror> methodParams = method.getParameterTypes();
+
+        /* Fetching the return type of the method in the format ASM
+         * parses the return type in a method descriptor.
+         */
+        TypeMirror returnType = method.getReturnType();
+        String returnTypeString = "";
+        if (returnType.getKind().equals(TypeKind.ARRAY)) {
+            returnTypeString = getArray(returnType);
+        } else {
+            returnTypeString = getParameter(returnType);
+        }
+
+        String methodArguments = "";
+
+        /* For an inner class constructor, adding the outer class object as an
+         * argument.
+         * Example:
+         * fullClassName = "org.gaul.example_package.OuterClass$InnerClass"
+         * methodArguments += "org.gaul.example_package.OuterClass"
+         */
+        String fullClassName =
+            getClassHeader(methodElement.getEnclosingElement());
+        if (methodElement.getKind().equals(ElementKind.CONSTRUCTOR)) {
+            int index = fullClassName.lastIndexOf("$");
```

Comment:
Does this still correct if it's a static inner class?

Reply:
No. This is only in case of non-static inner classes. Will change this to handle this case as well.